Google.PowerShell/Compute/GceMetadataCmdlet.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+
+using Google.PowerShell.Common;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Management.Automation;
+using System.Net;
+
+namespace Google.PowerShell.ComputeEngine
+{
+    /// <summary>
+    /// <para type="synopsis">
+    /// Gets the current metadata from the metadata server.
+    /// </para>
+    /// <para type="description">
+    /// Gets the current metadata from the metadata server. Get-GceMetadata can only be called from a Google
+    /// Compute Engine VM instance. Calls from any other machine will fail.
+    /// </para>
+    /// <example>
+    /// <code>PS C:\> $allMetadata = Get-GceMetadata | ConvertFrom-Json</code>
+    /// <para>Gets all of the metadata and converts it from the given json to a PSCustomObject.</para>
+    /// </example>
+    /// <example>
+    /// <code>PS C:\> $hostName = Get-GceMetadata -Path "instance/hostname" -NotRecursive</code>
+    /// <para>Gets the hostname of the instance.</para>
+    /// </example>
+    /// <example>
+    /// <code>
+    ///     PS C:\> $newTags, $newEtag = Get-GceMetadata -Path "instance/tags" -AppendEtag -WaitUpdate `
+    ///         -LastETag $oldETag
+    /// </code>
+    /// </example>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "GceMetadata")]
+    [OutputType(typeof(string))]
+    public class GetGceMetadataCmdlet : GCloudCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The path to the specific metadata you wish to get e.g. "instance/tags", "instance/attributes",
+        /// "project/attributes/sshKeys".
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public string Path { get; set; } = "";
+
+        /// <summary>
+        /// <para type="description">
+        /// If set, will only get the direct children of the current path.
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public SwitchParameter NotRecursive { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// When set, the value of the respone etag will be appended to the output pipeline after the content.
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public SwitchParameter AppendETag { get; set; }
```

Comment:
Can you provide an example of how you can actually use this? Because it doesn't seem possible.

    $metadataValue = Get-GceMetadata ... -AppendETag

Will `$metadataValue` have the value of the metadata or the ETag value? Is it possible to get both, a la:

    $metadata, $etag = Get-GceMetdata ... -AppendETag

If not, then perhaps we should have it return a dictionary in that case.

If `-AppendETag` is not set, just return a string. Otherwise return a dictionary: `@{ "Value"=...; "ETag"=...}`.